File: libs/nodejs/src/embeddings.ts
import type { HttpClient } from "./client.js";
import type { EmbedOptions, EmbedResult } from "./types.js";

export function createEmbeddingMethods(http: HttpClient) {
	return {
		async embed(
			bitId: string,
			input: string | string[],
			options?: EmbedOptions,
		): Promise<EmbedResult> {
			const texts = Array.isArray(input) ? input : [input];
			return http.request<EmbedResult>("POST", "/embeddings/embed", {
				body: {
					model: bitId,
					input: texts,
					embed_type: options?.embed_type ?? "query",
				},
				signal: options?.signal,
			});
		},
	};
}

File: libs/python/flow_like/embeddings.py
from __future__ import annotations

from typing import Any, Literal

from ._http import HTTPClient
from ._types import EmbeddingResult


class EmbeddingsMixin(HTTPClient):
    def embed(
        self,
        bit_id: str,
        input: str | list[str],
        embed_type: Literal["query", "document"] = "query",
        **kwargs: Any,
    ) -> EmbeddingResult:
        texts = [input] if isinstance(input, str) else input
        payload: dict[str, Any] = {
            "model": bit_id,
            "input": texts,
            "embed_type": embed_type,
            **kwargs,
        }
        resp = self._request("POST", "/embeddings/embed", json=payload)
        data = resp.json()
        return EmbeddingResult(
            embeddings=data.get("embeddings", []),
            usage=data.get("usage", {}),
            raw=data,
        )

    async def aembed(
        self,
        bit_id: str,
        input: str | list[str],
        embed_type: Literal["query", "document"] = "query",
        **kwargs: Any,
    ) -> EmbeddingResult:
        texts = [input] if isinstance(input, str) else input
        payload: dict[str, Any] = {
            "model": bit_id,
            "input": texts,
            "embed_type": embed_type,
            **kwargs,
        }
        resp = await self._arequest("POST", "/embeddings/embed", json=payload)
        data = resp.json()
        return EmbeddingResult(
            embeddings=data.get("embeddings", []),
            usage=data.get("usage", {}),
            raw=data,
        )


__all__ = ["EmbeddingsMixin"]
